### PR TITLE
 Master port to OM - OutputFilter

### PR DESCRIPTION
**Pull request by s7design**
Hello,

Here's PR with ported OutputFilter's modules. During porting and creating Selenium test for OutputFilterTextAutoLink.pm, we stumble upon problem. This module did not work for us when activated.
After some thorough research and testing of module, we found out, what we think is possible cause of bug. RegExp is correctly loaded from SysConfig, but it seems that at line 76 it is not assigned correctly further to MatchData. We tried to bypass it by hardcoding RegExp there and assigning it to MatchData, which worked partially. What we tried is:  replacing "(m{([\s:]$RegExp)}gxi)" with just "$RegExp". In AgentTicketZoom it did indeed matched and inserted appropriate images and links but in the middle of matched text. This might be totally wrong approach and way, but it was one of many ideas we tried around during testing.

Our field of expertise with RegExp (will work on improving it), limit us further investigating cause of problem, hopefully you can help us there and point in right direction.

It seems that this issue is long known to you, it is reported in bug #3923 and been popping up over the years by various bug reporters. Please just let us know if this module works ok for you, or is it work in progress.

King regards,
Sanjin
S7

### CODE REVIEW

**Comment by mgruner** [comment]
Hi Sanjin, I merged your changes, thank you. I didn't look into the bug yet, for now just porting is ok.

**Comment by mgruner** [comment]
Hello, I decided to revert this. As you can see we have different types of output filters: http://otrs.github.io/doc/manual/developer/stable/en/html/otrs-module-layers.html#outputfilter. I'd recommend that you change the name space to Output/HTML/FilterText for these filters. Can you please update the PR accordingly?

**Comment by mgruner** [comment]
Sorry for the misinformation in the first place.

**Comment by s7design** [comment]
Hi @mgruner 

It is not a problem, we will do that soon.

Regards
Zoran

**Comment by s7design** [comment]
Hi @mgruner 

PR is updated with changed name space.

Regards,
Sanjin

File: Kernel/Output/HTML/FilterText/TextAutoLink.pm [hunk 1]
```diff
@@ -6,11 +6,13 @@
 # did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
```
Comment (mgruner) [inline Kernel/Output/HTML/FilterText/TextAutoLink.pm]: I think we don't need Text twice. FilterText/AutoLink would be enough.

**Comment by s7design** [comment]
Hi @mgruner 

Think it's better like this as well, PR is updated.

Regards,
Sanjin

**Comment by mgruner** [comment]
Merged, thank you Sanjin!

**Pull request closed without merging**